### PR TITLE
Add static `update` method

### PR DESCRIPTION
**Pull request by kyleconroy**
You can now update API resources with first calling retrieve.

### CODE REVIEW

File: lib/ApiResource.php [hunk 1]
```diff
@@ -64,11 +64,10 @@ public static function classUrl()
     }
 
     /**
-     * @return string The full API URL for this API resource.
+     * @return string The instance endpoint URL for the given class.
      */
-    public function instanceUrl()
+    public static function classInstanceUrl($id)
```
Comment (brandur) [inline lib/ApiResource.php]: I _think_ you could just call this `instanceUrl` as well. The disambiguation is kind of nice, but I'm not crazy about starting a convention to prefix static functions with `class*`.

File: lib/ApiResource.php [hunk 2]
```diff
@@ -151,6 +158,18 @@ protected static function _create($params = null, $options = null)
         return $obj;
     }
 
+    protected static function _update($id, $params = null, $options = null)
```
Comment (brandur) [inline lib/ApiResource.php]: Just in the interest of leaving things better than we found them, mind putting some documentation on this function?

**Comment by brandur** [comment]
@kyleconroy Left a couple minor comments above, but overall looks good to me. Thanks!

File: lib/ApiResource.php (hunk 2 above)
Comment (kyleconroy) [inline lib/ApiResource.php]: +1 can do

File: lib/ApiResource.php (hunk 1 above)
Comment (kyleconroy) [inline lib/ApiResource.php]: Sadly you can't have a static function and non-static function with the same name.

File: lib/ApiResource.php [hunk 3]
```diff
@@ -64,11 +64,10 @@ public static function classUrl()
     }
 
     /**
-     * @return string The full API URL for this API resource.
+     * @return string The instance endpoint URL for the given class.
      */
-    public function instanceUrl()
+    public static function resourceUrl($id)
```
Comment (brandur) [inline lib/ApiResource.php]: +1 on the new name! Thanks.

**Comment by brandur** [comment]
@kyleconroy LGTM!

Let me know if you have any particular ideas around how/when you want to bring this in. Otherwise, I can coordinate a release if you like.

**Comment by kyleconroy** [comment]
I was planning on merging to master and doing a release. Is there anything else that needs to go out?

**Comment by brandur** [comment]
> I was planning on merging to master and doing a release. Is there anything else that needs to go out?

Nope! Aside from these changes, master should be clean right now.

File: lib/ApiResource.php [hunk 4]
```diff
@@ -151,6 +158,25 @@ protected static function _create($params = null, $options = null)
...
+     * @param string $id The ID of the API resource to update.
+     * @param array|null $params
+     * @param array|string|null $opts
+     *
+     * @return ApiResource the updated API resource
+     */
+    protected static function _update($id, $params = null, $options = null)
+    {
+        self::_validateParams($params);
+        $base = static::baseUrl();
```
Comment (arcanedev-maroc) [inline lib/ApiResource.php]: I think you can remove this line

**Pull request merged**